### PR TITLE
Section Feature Selection with wrapper method

### PR DESCRIPTION
**Pull request by be-marc**
Merge after mlr3viz and mlr3proba CRAN update and https://github.com/mlr-org/mlr3/issues/552 fix.

### CODE REVIEW

File: bookdown/03-optimization-feature-selection.Rmd [hunk 1]
```diff
@@ -109,4 +109,171 @@ Work in progress.
...
+  task = task,
+  learner = learner,
+  resampling = hout,
+  measure = measure,
+  terminator = evals20
+)
+instance
+```
+
+To start the feature selecton, we still need to select a feature selection algorithm which are defined via the `r ref("FSelector")` class.
```
Comment (RaphaelS1) [inline bookdown/03-optimization-feature-selection.Rmd:166]: typo, should be: "To start the feature selection"

Comment (RaphaelS1) [reply bookdown/03-optimization-feature-selection.Rmd:166]: though would probably read better as: "To start the feature selection, we still need to select an algorithm which are defined via the `r ref("FSelector")` class."

File: bookdown/03-optimization-feature-selection.Rmd [hunk 2]
```diff
@@ -109,4 +109,171 @@ Work in progress.
...
+
+In this example, we will use a simple random search.
+
+```{r 03-optimization-feature-selection-010}
+fselector = fs("random_search")
+```
+
+### Triggering the Tuning {#wrapper-selection-triggering}
+
+To start the feature selection, we simply pass the `r ref("FSelectInstanceSingleCrit")` to the `$optimize()` method of the initialized `r ref("FSelector")`. The algortihm proceeds as follows:
```
Comment (RaphaelS1) [inline bookdown/03-optimization-feature-selection.Rmd:186]: typo should be: "The algorithm proceeds as follows"

File: bookdown/03-optimization-feature-selection.Rmd [hunk 3]
```diff
@@ -109,4 +109,171 @@ Work in progress.
...
+
+1. The `r ref("FSelector")` proposes at least one feature subset  (the `r ref("FSelector")` may propose multiple subsets to improve parallelization, which can be controlled via the setting `batch_size`).
```
Comment (RaphaelS1) [inline bookdown/03-optimization-feature-selection.Rmd:188]: this reads slightly confusingly to me, how about: "The `r ref("FSelector")` proposes at least one feature subset and may propose multiple subsets to improve parallelization, which can be controlled via the setting `batch_size`."

File: bookdown/03-optimization-feature-selection.Rmd [hunk 4]
```diff
@@ -109,4 +109,171 @@ Work in progress.
...
+2. For each feature subset, the given `r ref("Learner")` is fitted on the `r ref("Task")` using the provided `r ref("Resampling")`.
+   All evaluations are stored in the archive of the `r ref("FSelectInstanceSingleCrit")`.
+3. The `r ref("Terminator")` is queried if the budget is exhausted.
+   If the budget is not exhausted, restart with 1) until it is.
+4. Determine the feature subset with the best observed performance.
+5. Store the best feature subset as result in the instance object.
```
Comment (RaphaelS1) [inline bookdown/03-optimization-feature-selection.Rmd:194]: possible typo: "Store the best feature subset as **a** result"

File: bookdown/03-optimization-feature-selection.Rmd [hunk 5]
```diff
@@ -109,4 +109,171 @@ Work in progress.
...
+The best featue subset (`$result_feature_set`) and the corresponding measured performance (`$result_y`) can be accessed from the instance.
```
Comment (RaphaelS1) [inline bookdown/03-optimization-feature-selection.Rmd:195]: typo, "The best feature subset "

File: bookdown/03-optimization-feature-selection.Rmd [hunk 6]
```diff
@@ -109,4 +109,171 @@ Work in progress.
...
+```{r 03-optimization-feature-selection-014}
+task$select(instance$result_feature_set)
+learner$train(task)
+```
+
+The trained model can now be used to make a prediction on external data.
+Note that predicting on observations present in the `task`,  should be avoided.
+The model has seen these observations already during feature selection and therefore results would be statistically biased.
+Hence, the resulting performance measure would be over-optimistic.
+Instead, to get statistically unbiased performance estimates for the current task, [nested resampling](#nested-resamling) is required.
```
Comment (RaphaelS1) [inline bookdown/03-optimization-feature-selection.Rmd:227]: typo, "#nested-resampling"

File: bookdown/03-optimization-feature-selection.Rmd [hunk 7]
```diff
@@ -109,4 +109,171 @@ Work in progress.
...
+
+### Automating the Feature Selection {#autofselect}
+
+The `r ref("AutoFSelect")` wraps a learner and augments it with an automatic feature selection for a given task.
+Because the `r ref("AutoFSelect")` itself inherits from the `r ref("Learner")` base class, it can be used like any other learner.
+Analogously to the previous subsection, a new classification tree learner is created.
+This classification tree learner automatically starts a feature selection on the given task using an inner resampling (holdout).
+We create a terminator which allows 10 evaluations, and use a simple random search as feature selection algorithm:
```
Comment (RaphaelS1) [inline bookdown/03-optimization-feature-selection.Rmd:235]: "and uses a simple random..."

File: bookdown/03-optimization-feature-selection.Rmd [hunk 8]
```diff
@@ -109,4 +109,171 @@ Work in progress.
...
+  resampling = rsmp("holdout"),
+  measure = msr("classif.ce"),
+  terminator = terminator,
+  fselector = fselector
+)
+at
+```
+
+We can now use the learner like any other learner, calling the `$train()` and `$predict()` method.
+This time however, we pass it to `r ref("benchmark()")` to compare the optimized featue subset to the complete feature set.
```
Comment (RaphaelS1) [inline bookdown/03-optimization-feature-selection.Rmd:256]: "to compare the optimized feature subset to the complete feature set"

**Review by RaphaelS1 — COMMENTED** [review]
@be-marc  I mainly checked for typos, didn't try building it but let me know if you want me to.

**Comment by be-marc** [comment]
@RaphaelS1 Thanks for your review!

**Pull request merged**